usautobuild/config.py

```diff
@@ -1,127 +1,171 @@
-        "CDN_PASSWORD": "cdn_password",
-        "DOCKER_PASSWORD": "docker_password",
-        "DOCKER_USERNAME": "docker_username",
-        "CHANGELOG_API_URL": "changelog_api_url",
-        "CHANGELOG_API_KEY": "changelog_api_key",
-    }
-    config_to_envs_map = {v: k for k, v in envs_to_config_map.items()}
-
-    cdn_host: str
-    cdn_user: str
-    cdn_password: str
-    docker_password: str
-    docker_username: str
-    changelog_api_url: str
-    changelog_api_key: str
-
-    config_file = None
-    git_url = "https://github.com/unitystation/unitystation.git"
-    git_branch = "develop"
-    allow_no_changes = True
-    unity_version = "2020.1.17f1"
-    target_platforms = ["linuxserver", "StandaloneWindows64", "StandaloneOSX", "StandaloneLinux64"]
-    cdn_download_url = "https://unitystationfile.b-cdn.net/{}/{}/{}.zip"
-    forkname = "UnityStationDevelop"
-    output_dir = Path.cwd() / "builds"
-    license_file = Path.cwd() / "UnityLicense.ulf"
-    discord_webhook = None
-    abort_on_build_fail = True
-
-    project_path = Path()
-    build_number = 0
-
-    def __init__(self, args: dict[str, Any], config_file: Optional[str] = None):
-        self.args = args
-        if config_file:
-            self.config_file = Path(config_file)
-        self.handle_config_file()
-        self.handle_args()
-        self.set_required_envs()
-
-    def set_required_envs(self) -> None:
-        log.info("Setting required envs...")
-        for env_key, config_key in self.envs_to_config_map.items():
-            value = os.environ.get(env_key)
-            if value is None:
-                log.error(f"Required env is missing: {env_key}")
-                raise MissingRequiredEnv(env_key)
-
-            setattr(self, config_key, value)
-
-    def handle_config_file(self) -> None:
-        if self.config_file is None:
-            config_json = Path("config.json")
-            if config_json.is_file():
-                self.config_file = config_json
-
-        if self.config_file is None:
+    cdn_host: str = Var()
+    cdn_user: str = Var()
+    cdn_password: str = Var()
+    docker_password: str = Var()
+    docker_username: str = Var()
+    changelog_api_url: str = Var()
+    changelog_api_key: str = Var()
+
+    git_url = Var("https://github.com/unitystation/unitystation.git")
+    git_branch = Var("develop")
+
+    unity_version = Var("2020.1.17f1")
+    target_platforms = Var(["linuxserver", "StandaloneWindows64", "StandaloneOSX", "StandaloneLinux64"])
+    cdn_download_url = Var("https://unitystationfile.b-cdn.net/{}/{}/{}.zip")
+    forkname = Var("UnityStationDevelop")
+
+    discord_webhook: Optional[str] = Var(None)
+
+    dry_run = Var(False)
+    abort_on_build_fail = Var(True)
+    allow_no_changes = Var(True)
+
+    build_number = Var(0)
+
+    output_dir = Var(Path.cwd() / "builds")
+    license_file = Var(Path.cwd() / "UnityLicense.ulf")
+    project_path = Var(Path())
+
+    def __init__(self, args: dict[str, Any]):
```

Comment:
this init method is chonky, isn't it?

Reply:
It's not super complex. I agree that it's a good idea to put code somewhere else but i had no time today

Reply:
fixed